pkg/driftctl_test.go

```diff
@@ -822,51 +822,55 @@ func TestDriftctlRun_Middlewares(t *testing.T) {
 		{
 			name: "test sns topic policy expander middleware",
 			stateResources: []resource.Resource{
-				&aws.AwsSnsTopic{
-					Id:     "foo",
-					Arn:    awssdk.String("arn"),
-					Policy: awssdk.String("{\"policy\":\"bar\"}"),
+				&resource.AbstractResource{
+					Id:   "foo",
+					Type: aws.AwsSnsTopicResourceType,
+					Attrs: &resource.Attributes{
+						"arn":    "arn",
+						"id":     "foo",
+						"policy": "{\"policy\":\"bar\"}",
+					},
 				},
 			},
 			remoteResources: []resource.Resource{
-				&aws.AwsSnsTopicPolicy{
-					Id:     "foo",
-					Arn:    awssdk.String("arn"),
-					Policy: awssdk.String("{\"policy\":\"baz\"}"),
-					CtyVal: func() *cty.Value {
-						v := cty.ObjectVal(map[string]cty.Value{
-							"id":     cty.StringVal("foo"),
-							"arn":    cty.StringVal("arn"),
-							"policy": cty.StringVal("{\"policy\":\"baz\"}"),
-						})
-						return &v
-					}(),
+				&resource.AbstractResource{
+					Id:   "foo",
+					Type: aws.AwsSnsTopicPolicyResourceType,
+					Attrs: &resource.Attributes{
+						"id":     "foo",
+						"arn":    "arn",
+						"policy": "{\"policy\":\"baz\"}",
+					},
 				},
 			},
 			mocks: func(factory resource.ResourceFactory) {
-				foo := cty.ObjectVal(map[string]cty.Value{
-					"id":     cty.StringVal("foo"),
-					"arn":    cty.StringVal("arn"),
-					"policy": cty.StringVal("{\"policy\":\"bar\"}"),
-				})
-				factory.(*terraform.MockResourceFactory).On("CreateResource", mock.MatchedBy(func(input map[string]interface{}) bool {
+				factory.(*terraform.MockResourceFactory).On("CreateAbstractResource", "foo", "aws_sns_topic_policy", mock.MatchedBy(func(input map[string]interface{}) bool {
```

Comment:
`mock.MatchedBy` can be removed and you can use directly a `map[string]interface{}`